share/wrf_tsin.F

```diff
@@ -46,17 +49,35 @@ SUBROUTINE wrf_tsin ( grid , ierr )
    
              IF (istatus == 0) THEN
    
-                ! Ignore first three lines, which constitute a header
-                READ(UNIT=iunit, FMT='(1X)')
+                ! The header will be parsed to see if lat/lon or i/j is given...
                 READ(UNIT=iunit, FMT='(1X)')
+                READ(UNIT=iunit, FMT='(A32,A7,A2,A7,A1)') head1, JorLat, head2, IorLon, head3
                 READ(UNIT=iunit, FMT='(1X)')
+                grid%tslist_ij = .false.
+                ! If the header contains "LAT" and "LON" then it will expect
+                !... lat/lon coords in the file
+                ! If the header has "J" and "I", then i,j will be used.
+                if(index(JorLat,'LAT').ne.0 .and. index(IorLon,'LON').ne.0) then
+                    grid%tslist_ij = .false.
+                elseif(index(JorLat,'J').ne.0 .and. index(IorLon,'I').ne.0) then
+                    grid%tslist_ij = .true.
+                endif
```

Comment:
It may be nice to handle the case where creative users write these in lower-case letters, or in mixed-case.

@davegill Does WRF already provide a standard facility for capitalizing a string? If not, we could add something like the following to an appropriate place in WRF:
```fortran
    function capitalize(str) result(capStr)
!<DESCRIPTION>
!
! Returns a copy of 'str' in which all lower-case letters have been converted
! to upper-case letters.
!
!</DESCRIPTION>

        character(len=*), intent(in) :: str
        character(len=len(str)) :: capStr

        integer :: i
        integer, parameter :: offset = (iachar('a') - iachar('A'))


        do i=1,len(str)
            if (iachar(str(i:i)) >= iachar('a') .and. iachar(str(i:i)) <= iachar('z')) then
               capStr(i:i) = achar(iachar(str(i:i)) - offset)
            else
               capStr(i:i) = str(i:i)
            end if
        end do

    end function capitalize
```
If we go this route, I can provide a version that has capitalized Fortran keywords to match the rest of the WRF infrastructure.

Reply:
@mgduda @kkeene44 @phawbeck 
Folks,
I have implemented Michael's suggestion (and if you look at the code, you'll see how closely I took his suggestion). In the share directory is a module_string_tools.F This has a single function (right now) that turns an input string into an upper case string.